plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/rows/update.tsx

```diff
@@ -0,0 +1,127 @@
+/**
+ * External dependencies
+ */
+import { Button, Icon } from '@wordpress/components';
+import { __, sprintf } from '@wordpress/i18n';
+import { useContext, useState } from '@wordpress/element';
+import { useDispatch } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { Subscription } from '../../types';
+import { SubscriptionsContext } from '../../../../contexts/subscriptions-context';
+
+interface UpdateProps {
+	subscription: Subscription;
+}
+
+export default function Update( props: UpdateProps ) {
+	const [ isUpdating, setIsUpdating ] = useState( false );
+	const { createWarningNotice, createSuccessNotice } =
+		useDispatch( 'core/notices' );
+	const { loadSubscriptions } = useContext( SubscriptionsContext );
+
+	const canUpdate = props.subscription.active && props.subscription.local;
+
+	function update() {
+		if ( ! canUpdate ) {
+			return;
```

Comment:
Should we run `setIsUpdating` to true here so that the button is disabled or disable the button another way if updating isn't available?

Reply:
No, this check prevents the update event from triggering in case the update is impossible. 

In a future PR I will add support for modals like buy and activate, so the update button will always be clickable, but it may not run the update action if not available.